From 952c6b9afb47e4c4f4a63261d6611b48238b2b22 Mon Sep 17 00:00:00 2001
From: Chad Metcalf <chad@cloudera.com>
Date: Fri, 25 Jun 2010 23:25:56 -0700
Subject: [PATCH 1/5] CLOUDERA-BUILD. Adding basic cloudera build infrastructure support.
 - updates include docs in build
 - Install docs dir (non gpl), man page dir.

---
 build.xml                 |    2 +-
 cloudera/README.cloudera  |    9 +++
 cloudera/do-release-build |   37 ++++++++++++
 cloudera/install_flume.sh |  136 +++++++++++++++++++++++++++++++++++++++++++++
 src/saveVersion.sh        |    2 +-
 5 files changed, 184 insertions(+), 2 deletions(-)
 create mode 100644 cloudera/README.cloudera
 create mode 100755 cloudera/do-release-build
 create mode 100644 cloudera/install_flume.sh

diff --git a/build.xml b/build.xml
index f1b595b..c02db3f 100644
--- a/build.xml
+++ b/build.xml
@@ -429,7 +429,7 @@ to call at top-level: ant deploy-contrib compile-core-test
       of the output directory -->
     <exec executable="sh">
        <arg line="src/saveVersion.sh ${version}"/>
-      <!--  <env key="HADOOP_REVISION" value="${cloudera.hash}" /> -->
+      <env key="FLUME_REVISION" value="${cloudera.hash}" /> 
     </exec>
     <javac 
      encoding="${build.encoding}" 
diff --git a/cloudera/README.cloudera b/cloudera/README.cloudera
new file mode 100644
index 0000000..8626e8f
--- /dev/null
+++ b/cloudera/README.cloudera
@@ -0,0 +1,9 @@
+This build was generated by Cloudera's build system in the following manner:
+
+1) The pristine open-source release tarball was unpacked
+
+2) The patches contained within the patches/ directory next to this README
+were applied using the apply-patches script. A complete log of these changes
+is also included in CHANGES.cloudera.txt.
+
+3) The project was built by running the do-release-build script in this directory.
diff --git a/cloudera/do-release-build b/cloudera/do-release-build
new file mode 100755
index 0000000..6c2981b
--- /dev/null
+++ b/cloudera/do-release-build
@@ -0,0 +1,37 @@
+#!/bin/bash
+# Copyright (c) 2009 Cloudera, inc
+#
+# Performs a release build
+
+set -ex
+
+JAVA32_HOME=${JAVA32_HOME:-$JAVA_HOME}
+
+# Check that JVMs support the right architectures
+# Passing -d with an architecture that's not supported results
+# in a non-zero exit code even with -version
+if ! $JAVA32_HOME/bin/java -d32 -version ; then
+  echo Your jvm in $JAVA32_HOME seems like it doesnt support 32-bit mode 1>&2
+  echo Please set JAVA32_HOME to point to a 32-bit JDK
+  exit 1
+fi
+
+if ! [ -e $JAVA5_HOME/bin/java ]; then
+  echo No JAVA5_HOME set. Forrest documentation requires Java 5 to be 1>&2
+  echo installed. 1>&2
+  exit 1
+fi
+
+if ! [ -e $FORREST_HOME/bin/forrest ]; then
+  echo No FORREST_HOME set. Forrest documentation requires Forrest to be 1>&2
+  echo installed. 1>&2
+  exit 1
+fi
+
+# Do the build
+BIN_DIR=$(readlink -f $(dirname $0))
+RELEASE_DIR=$BIN_DIR/..
+
+cd $RELEASE_DIR
+
+JAVA_HOME=$JAVA32_HOME  ant clean tar -Dversion=${FULL_VERSION} -propertyfile cloudera/build.properties 
diff --git a/cloudera/install_flume.sh b/cloudera/install_flume.sh
new file mode 100644
index 0000000..c15a7fb
--- /dev/null
+++ b/cloudera/install_flume.sh
@@ -0,0 +1,136 @@
+#!/bin/sh
+# Copyright 2009 Cloudera, inc.
+set -ex
+
+usage() {
+  echo "
+usage: $0 <options>
+  Required not-so-options:
+     --cloudera-source-dir=DIR   path to cloudera distribution files
+     --build-dir=DIR             path to flumedist.dir
+     --prefix=PREFIX             path to install into
+
+  Optional options:
+     --doc-dir=DIR               path to install docs into [/usr/share/doc/flume]
+     --lib-dir=DIR               path to install flume home [/usr/lib/flume]
+     --installed-lib-dir=DIR     path where lib-dir will end up on target system
+     --bin-dir=DIR               path to install bins [/usr/bin]
+     --examples-dir=DIR          path to install examples [doc-dir/examples]
+     ... [ see source for more similar options ]
+  "
+  exit 1
+}
+
+OPTS=$(getopt \
+  -n $0 \
+  -o '' \
+  -l 'cloudera-source-dir:' \
+  -l 'prefix:' \
+  -l 'doc-dir:' \
+  -l 'lib-dir:' \
+  -l 'installed-lib-dir:' \
+  -l 'bin-dir:' \
+  -l 'examples-dir:' \
+  -l 'build-dir:' -- "$@")
+
+if [ $? != 0 ] ; then
+    usage
+fi
+
+eval set -- "$OPTS"
+set -ex
+while true ; do
+    case "$1" in
+        --cloudera-source-dir)
+        CLOUDERA_SOURCE_DIR=$2 ; shift 2
+        ;;
+        --prefix)
+        PREFIX=$2 ; shift 2
+        ;;
+        --build-dir)
+        BUILD_DIR=$2 ; shift 2
+        ;;
+        --doc-dir)
+        DOC_DIR=$2 ; shift 2
+        ;;
+        --lib-dir)
+        LIB_DIR=$2 ; shift 2
+        ;;
+        --installed-lib-dir)
+        INSTALLED_LIB_DIR=$2 ; shift 2
+        ;;
+        --bin-dir)
+        BIN_DIR=$2 ; shift 2
+        ;;
+        --examples-dir)
+        EXAMPLES_DIR=$2 ; shift 2
+        ;;
+        --)
+        shift ; break
+        ;;
+        *)
+        echo "Unknown option: $1"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+for var in CLOUDERA_SOURCE_DIR PREFIX BUILD_DIR ; do
+  if [ -z "$(eval "echo \$$var")" ]; then
+    echo Missing param: $var
+    usage
+  fi
+done
+
+MAN_DIR=${MAN_DIR:-/usr/share/man/man1}
+DOC_DIR=${DOC_DIR:-/usr/share/doc/flume}
+LIB_DIR=${LIB_DIR:-/usr/lib/flume}
+BIN_DIR=${BIN_DIR:-/usr/lib/flume/bin}
+CONF_DIR=/etc/flume/
+CONF_DIST_DIR=/etc/flume/conf.dist/
+ETC_DIR=${ETC_DIR:-/etc/flume}
+
+
+install -d -m 0755 ${PREFIX}/${LIB_DIR}
+
+install -d -m 0755 ${PREFIX}/${LIB_DIR}/lib
+for i in `find lib/*.jar build/lib/*.jar -type f |grep -v zookeeper`
+	do echo "Copying $i"
+	cp $i ${PREFIX}/${LIB_DIR}/lib #don't copy directories by default
+done
+
+cp flume*.jar ${PREFIX}/${LIB_DIR}/lib
+
+cp -a webapps ${PREFIX}/${LIB_DIR}
+
+install -d -m 0755 $PREFIX/$BIN_DIR
+cp bin/* $PREFIX/${BIN_DIR}
+
+wrapper=$PREFIX/usr/bin/flume
+mkdir -p `dirname $wrapper`
+cat > $wrapper <<EOF
+#!/bin/sh
+
+exec /usr/lib/flume/bin/flume "\$@"
+EOF
+chmod 755 $wrapper
+
+
+install -d -m 0755 $PREFIX/$ETC_DIR/conf.empty
+(cd ${BUILD_DIR}/conf && tar cf - .) | (cd $PREFIX/$ETC_DIR/conf.empty && tar xf -)
+
+unlink $PREFIX/$LIB_DIR/conf || /bin/true
+ln -s /etc/flume/conf $PREFIX/$LIB_DIR/conf
+
+# Docs
+install -d -m 0755 $PREFIX/${DOC_DIR}
+cp -r docs/* $PREFIX/${DOC_DIR}
+
+
+# man pages
+install -d -m 0755 $PREFIX/$MAN_DIR
+gzip -c $CLOUDERA_SOURCE_DIR/../docs/man/flume.1 > $PREFIX/$MAN_DIR/flume.1.gz
+gzip -c $CLOUDERA_SOURCE_DIR/../docs/man/flume-master.1 > $PREFIX/$MAN_DIR/flume-master.1.gz
+gzip -c $CLOUDERA_SOURCE_DIR/../docs/man/flume-node.1 > $PREFIX/$MAN_DIR/flume-node.1.gz
+gzip -c $CLOUDERA_SOURCE_DIR/../docs/man/flume-shell.1 > $PREFIX/$MAN_DIR/flume-shell.1.gz
diff --git a/src/saveVersion.sh b/src/saveVersion.sh
index 502d509..74dbaa6 100755
--- a/src/saveVersion.sh
+++ b/src/saveVersion.sh
@@ -19,7 +19,7 @@
 # This file is used to generate the BuildStamp.java class that
 # records the user, url, revision and timestamp.
 version=$1
-revision=$HADOOP_REVISION
+revision=$FLUME_REVISION
 #if [ -z "$revision" ] ; then
 #  revision=`svn info | sed -n -e 's/Last Changed Rev: \(.*\)/\1/p'`
 #fi
-- 
1.7.0.4

